shell/platform/darwin/ios/ios_surface_gl.mm

```diff
@@ -6,49 +6,49 @@
 
 #include "flutter/fml/trace_event.h"
 #include "flutter/shell/gpu/gpu_surface_gl.h"
+#include "flutter/shell/platform/darwin/ios/ios_context_gl.h"
 
 namespace flutter {
 
-IOSSurfaceGL::IOSSurfaceGL(std::shared_ptr<IOSGLContext> context,
-                           fml::scoped_nsobject<CAEAGLLayer> layer,
-                           FlutterPlatformViewsController* platform_views_controller)
-    : IOSSurface(platform_views_controller), context_(context) {
-  render_target_ = context_->CreateRenderTarget(std::move(layer));
+static IOSContextGL* CastToGLContext(const std::shared_ptr<IOSContext>& context) {
+  return reinterpret_cast<IOSContextGL*>(context.get());
 }
 
-IOSSurfaceGL::IOSSurfaceGL(fml::scoped_nsobject<CAEAGLLayer> layer,
-                           std::shared_ptr<IOSGLContext> context)
-    : IOSSurface(nullptr), context_(context) {
-  render_target_ = context_->CreateRenderTarget(std::move(layer));
+IOSSurfaceGL::IOSSurfaceGL(fml::scoped_nsobject<CALayer> layer,
+                           std::shared_ptr<IOSContext> context,
+                           FlutterPlatformViewsController* platform_views_controller)
+    : IOSSurface(context, platform_views_controller) {
+  render_target_ = CastToGLContext(context)->CreateRenderTarget(std::move(layer));
 }
 
 IOSSurfaceGL::~IOSSurfaceGL() = default;
 
+// |IOSSurface|
 bool IOSSurfaceGL::IsValid() const {
   return render_target_->IsValid();
 }
 
-bool IOSSurfaceGL::ResourceContextMakeCurrent() {
-  return context_->ResourceMakeCurrent();
-}
-
+// |IOSSurface|
 void IOSSurfaceGL::UpdateStorageSizeIfNecessary() {
   if (IsValid()) {
     render_target_->UpdateStorageSizeIfNecessary();
   }
 }
 
+// |IOSSurface|
 std::unique_ptr<Surface> IOSSurfaceGL::CreateGPUSurface(GrContext* gr_context) {
   if (gr_context) {
     return std::make_unique<GPUSurfaceGL>(sk_ref_sp(gr_context), this, true);
   }
   return std::make_unique<GPUSurfaceGL>(this, true);
 }
 
+// |GPUSurfaceGLDelegate|
 intptr_t IOSSurfaceGL::GLContextFBO() const {
-  return IsValid() ? render_target_->framebuffer() : GL_NONE;
+  return IsValid() ? render_target_->GetFramebuffer() : 0;
```

Comment:
Should this be -1?  I is zero an invalid FBO?  i can't remember now if counting starts at zero or one.

Reply:
```
#define GL_NONE 0
```

0 is an alias for default window-bound framebuffer on some platforms. I am not sure we we want it to be 0 as I'd prefer to default to -1 in case someone forgets to bind a framebuffer explicitly. But didn't want to change this behavior. Just didn't want to depend on OpenGL headers here but I see that we already depend on them here. So I am bringing back `GL_NONE`.